source/slang/check.cpp

```diff
@@ -9012,13 +8963,96 @@ namespace Slang
 
                 return createLookupResultExpr(
                     lookupResult,
-                    expr->BaseExpression,
+                    baseExpression,
                     expr->loc);
             }
             else if (as<ErrorType>(baseType))
             {
                 return CreateErrorExpr(expr);
             }
+
+            // Failure
+            return lookupMemberResultFailure(expr, baseType);
```

Comment:
This failure case here is kind of odd, because we are going to land in this case for something like:

```hlsl
int a = 5;
int b = a::something;
```

We will report to the user that `a` doesn't have a member named `something`, but that makes no sense as an error, because we should really be telling them that `a` isn't a type, and hence can't be used for looking up static members like this.

Reply:
I would expect a full-featured version of this feature to add a diagnostic for when we are doing static lookup into something that isn't appropriate for static lookup (in current Slang, that would be anything that isn't a type, but we might add `namespace` support later, which would add another case).